openverse_catalog/dags/providers/provider_api_scripts/nappy.py

```diff
@@ -0,0 +1,123 @@
+"""
+Content Provider:       Nappy
+
+ETL Process:            Use the API to identify all CC0-licensed images.
+
+Output:                 TSV file containing the image meta-data.
+
+Notes:                  This api was written specially for Openverse.
+                        There are no known limits or restrictions.
+
+"""
+import logging
+
+from common import constants
+from common.licenses import get_license_info
+from common.loader import provider_details as prov
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
+
+
+logger = logging.getLogger(__name__)
+
+
+class NappyDataIngester(ProviderDataIngester):
+    providers = {constants.IMAGE: prov.NAPPY_DEFAULT_PROVIDER}
+    endpoint = "https://api.nappy.co/v1/openverse/images"
+    headers = {"User-Agent": prov.UA_STRING, "Accept": "application/json"}
+
+    def get_next_query_params(self, prev_query_params: dict | None, **kwargs) -> dict:
+        # On the first request, `prev_query_params` will be `None`. We can detect this
+        # and return our default params.
+        if not prev_query_params:
+            return {
+                "page": 1,
+            }
+        else:
+            return {
+                "page": prev_query_params["page"] + 1,
+            }
+
+    def get_batch_data(self, response_json):
+        # Takes the raw API response from calling `get` on the endpoint, and returns
+        # the list of records to process.
+        if response_json:
+            return response_json.get("images")
+        return None
+
+    def get_should_continue(self, response_json):
+        return bool(response_json.get("next_page"))
+
+    def get_media_type(self, record: dict):
+        return constants.IMAGE
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        # Parse out the necessary info from the record data into a dictionary.
+
+        if (foreign_landing_url := data.get("foreign_landing_url")) is None:
+            return None
+
+        if (image_url := data.get("url")) is None:
+            return None
+
+        # Hardoded to CC0, the only license Nappy.co uses
+        license_info = get_license_info(
+            "https://creativecommons.org/publicdomain/zero/1.0/"
+        )
```

Comment:
If all results are CC0, we should set this as a value on the class or instance and use it there rather than calling this function for every record!

Reply:
Oops! Totally, yes.